realm/realm-library/src/main/java/io/realm/RealmList.java

```diff
@@ -218,15 +219,32 @@ public E set(int location, E object) {
 
     // Transparently copies a standalone object or managed object from another Realm to the Realm backing this RealmList.
     private E copyToRealmIfNeeded(E object) {
-        // Object is already in this realm
-        if (object.row != null && object.realm.getPath().equals(realm.getPath())) {
-            return object;
-        }
-
-        // We don't support moving DynamicRealmObjects across Realms automatically. The overhead is too big as you
-        // have to run a full schema validation for each object.
         if (object instanceof DynamicRealmObject) {
-            throw new IllegalArgumentException("Automatically copying DynamicRealmObjects from other Realms are not supported");
+            String listClassName = view.getTargetTable().getName().substring(Table.TABLE_PREFIX.length());
```

Comment:
I would prefer if this information was accessed through RealmSchema somehow. All these rules about naming should all be enforced there instead of being spread around the codebase (IMO). It probably means adding something like `getSchemaForTable` as I am not sure the functionality exists yet.